src/components/SiteTrackingRequired/SiteTrackingRequired.test.js

```diff
@@ -0,0 +1,119 @@
+import React from 'react';
+import { render, cleanup, waitForDomChange } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+import DopplerIntlProvider from '../../i18n/DopplerIntlProvider.double-with-ids-as-values';
+import { AppServicesProvider } from '../../services/pure-di';
+import { SiteTrackingRequired, SiteTrackingNotAvailableReasons } from './SiteTrackingRequired';
+
+describe('site tracking', () => {
+  afterEach(cleanup);
+
+  it('should show free account messages', () => {
+    // Arrange
+    const reason = SiteTrackingNotAvailableReasons.freeAccount;
+
+    // Act
+    const { getByText } = render(
+      <AppServicesProvider>
+        <DopplerIntlProvider>
+          <SiteTrackingRequired reason={reason} />
+        </DopplerIntlProvider>
+      </AppServicesProvider>,
+    );
+
+    // Assert
+    expect(getByText('reports.upgrade_account_free_title')).toBeInTheDocument();
+  });
+
+  it('should show trial not accepted messages', () => {
+    // Arrange
+    const reason = SiteTrackingNotAvailableReasons.trialNotAccepted;
+
+    // Act
+    const { getByText } = render(
+      <AppServicesProvider>
+        <DopplerIntlProvider>
+          <SiteTrackingRequired reason={reason} />
+        </DopplerIntlProvider>
+      </AppServicesProvider>,
+    );
+
+    // Assert
+    expect(getByText('reports.allow_enable_trial_title')).toBeInTheDocument();
+  });
+
+  it('should show not domains messages', () => {
+    // Arrange
+    const reason = SiteTrackingNotAvailableReasons.thereAreNotDomains;
+
+    // Act
+    const { getByText } = render(
+      <AppServicesProvider>
+        <DopplerIntlProvider>
+          <SiteTrackingRequired reason={reason} />
+        </DopplerIntlProvider>
+      </AppServicesProvider>,
+    );
+
+    // Assert
+    expect(getByText('reports.datahub_not_domains_title')).toBeInTheDocument();
+  });
+
+  it('should be activate the trial and redirect to control panel site tracking settings', async () => {
+    // Arrange
+    const reason = SiteTrackingNotAvailableReasons.trialNotAccepted;
+    const dependencies = {
+      dopplerLegacyClient: {
+        activateSiteTrackingTrial: async () => ({ success: true }),
+      },
+      appConfiguration: { dopplerLegacyUrl: 'localhost:3000' },
+      window: { location: { href: '/ControlPanel/CampaignsPreferences/SiteTrackingSettings' } },
```

Comment:
Realmente creo que setear este valor en el href confunde. Cual es la razón de setearlo?

Reply:
el double no era el que habia que pisar? para poder testear la url.. o te referis en el caso que sea false? porque comentaste el de true

Reply:
Yo modificaría esta línea así:

```suggestion
      window: { location: { href: '' } },
```

o 

```suggestion
      window: { location: { } },
```

Reply:
done